### PR TITLE
Stop using automatically generated service account token in e2e test suite

### PR DESCRIPTION
**Pull request by jrvaldes**
This change replaces the API call to fetch the service account token from the automatically generated secret with a TokenRequest.

As per official docs automatically generated secrets for should be avoided as they might be removed in a future OpenShift Container Platform release.

See https://docs.openshift.com/container-platform/4.14/authentication/using-service-accounts-in-applications.html#auto-generated-sa-token-secrets_using-service-accounts

### CODE REVIEW

File: test/e2e/metrics_test.go [hunk 1]
```diff
@@ -216,35 +217,16 @@ func (tc *testContext) testWindowsPrometheusRules(t *testing.T) {
...
-	var secretName string
-	for _, secret := range monitoringSecrets.Items {
-		if strings.Contains(secret.Name, "prometheus-k8s-token") {
-			secretName = secret.Name
-			break
-		}
+	tokenRequest := &auth.TokenRequest{
+		Spec: auth.TokenRequestSpec{},
 	}
+	token, err := tc.client.K8s.CoreV1().ServiceAccounts(monitoringNamespace).CreateToken(context.TODO(),
```
Comment (saifshaikh48) [inline test/e2e/metrics_test.go:223]: this creates a new token each time it is executed right? do the tokens expire, or do we have to revoke them somehow?

Comment (jrvaldes) [reply test/e2e/metrics_test.go:223]: > this creates a new token each time it is executed right? 

Yes

Comment (jrvaldes) [reply test/e2e/metrics_test.go:223]: > Do we have to revoke them somehow?

No.

> Do the tokens expire, 
Yes, the default ExpirationSeconds is 3600 seconds (1 hour)

**Comment by mansikulkarni96** [comment]
LGTM

File: test/e2e/metrics_test.go (hunk 1 above)
Comment (sebsoto) [reply test/e2e/metrics_test.go:223]: Please add the token expiration time to the function doc

**Review by sebsoto — CHANGES_REQUESTED** [review]
Thanks @jrvaldes, just have one request

**Pull request merged**